From d5b197439cb95eb79982c9a50edd4acab18d23d6 Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Sun, 19 Feb 2023 11:10:46 +0100
Subject: [PATCH] configure.ac: Add configure option for musicxml file support

musicxml file support requires lou_translate and Freedots programs,
where the latter is not available on some distributions. This
configuration option will adjust MIME convs and types in case a
maintainer can't ship non-functional filter when Freedots program is
missing.
---
 Makefile.am                              |  6 +++++-
 configure.ac                             | 20 +++++++++++++++++++-
 mime/{braille.convs => braille.convs.in} |  2 +-
 mime/{braille.types => braille.types.in} |  2 +-
 4 files changed, 26 insertions(+), 4 deletions(-)
 rename mime/{braille.convs => braille.convs.in} (98%)
 rename mime/{braille.types => braille.types.in} (96%)

diff --git a/Makefile.am b/Makefile.am
index d20df518..11a8454e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -147,6 +147,10 @@ endif
 # ======================
 genfilterscripts =
 pkgfilter_SCRIPTS = $(genfilterscripts)
+if ENABLE_MUSICXML
+musicxmlscripts = filter/musicxmltobrf
+endif
+
 if ENABLE_BRAILLE
 nodist_pkgfilter_SCRIPTS = \
 	driver/generic/brftoembosser \
@@ -158,7 +162,7 @@ nodist_pkgfilter_SCRIPTS = \
 	filter/vectortobrf \
 	filter/texttobrf \
 	filter/brftopagedbrf \
-	filter/musicxmltobrf
+	$(musicxmlscripts)
 endif
 
 distclean-local:
diff --git a/configure.ac b/configure.ac
index b1dd110a..cf73a070 100644
--- a/configure.ac
+++ b/configure.ac
@@ -188,6 +188,21 @@ PKG_CHECK_EXISTS([liblouis], [
 AM_CONDITIONAL(ENABLE_BRAILLE, test "x$enable_braille" = xyes)
 AC_SUBST(TABLESDIR)
 
+# ================
+# musicxml support
+# ================
+AC_ARG_ENABLE(musicxml, AS_HELP_STRING([--enable-musicxml],[enable musicxml file support, requires Freedots and lou_translate]),
+	      enable_musicxml=$enableval,enable_musicxml=no)
+if test "x$enable_musicxml" = "xyes"
+then
+	  MUSICXML_CONV="application/vnd.recordare.musicxml+xml application/vnd.cups-brf 30  musicxmltobrf"
+	  MUSICXML_TYPE="application/vnd.recordare.musicxml+xml xml contains(0,1000,\"<score-partwise\")"
+	  MUSICXML_FILTER="filter/musicxmltobrf"
+fi
+AM_CONDITIONAL(ENABLE_MUSICXML, test "x$enable_musicxml" = "xyes")
+AC_SUBST(MUSICXML_CONV)
+AC_SUBST(MUSICXML_TYPE)
+
 # =====================
 # Prepare all .in files
 # =====================
@@ -206,8 +221,10 @@ AC_CONFIG_FILES([
 	filter/brftopagedbrf
 	filter/vectortopdf
 	filter/vectortobrf
-	filter/musicxmltobrf
+	${MUSICXML_FILTER}
 	filter/liblouis1.defs.gen
+	mime/braille.convs
+	mime/braille.types
 ])
 AC_CONFIG_COMMANDS([executable-scripts], [
 	chmod +x filter/liblouis1.defs.gen
@@ -227,6 +244,7 @@ Build configuration:
 	cups-config:     ${with_cups_config}
 	braille:	 ${enable_braille}
 	braille tables:  ${TABLESDIR}
+	musicxml support:  ${enable_musicxml}
 	werror:          ${enable_werror}
 ==============================================================================
 ])
diff --git a/mime/braille.convs b/mime/braille.convs.in
similarity index 98%
rename from mime/braille.convs
rename to mime/braille.convs.in
index d741778e..e01c2dee 100644
--- a/mime/braille.convs
+++ b/mime/braille.convs.in
@@ -16,7 +16,7 @@ application/sgml	application/vnd.cups-brf	10	texttobrf
 application/vnd.cups-brf	application/vnd.cups-paged-brf	0	brftopagedbrf
 application/vnd.cups-ubrl	application/vnd.cups-paged-ubrl	0	brftopagedbrf
 
-application/vnd.recordare.musicxml+xml application/vnd.cups-brf	30	musicxmltobrf
+@MUSICXML_CONV@
 
 application/vnd.oasis.opendocument.chart			application/vnd.cups-brf	30	texttobrf
 application/vnd.oasis.opendocument.formula			application/vnd.cups-brf	30	texttobrf
diff --git a/mime/braille.types b/mime/braille.types.in
similarity index 96%
rename from mime/braille.types
rename to mime/braille.types.in
index f55ef2a8..295f96e3 100644
--- a/mime/braille.types
+++ b/mime/braille.types.in
@@ -16,7 +16,7 @@ application/vnd.cups-paged-ubrl
 
 application/sgml	sgml
 
-application/vnd.recordare.musicxml+xml xml contains(0,1000,"<score-partwise")
+@MUSICXML_TYPE@
 
 application/vnd.oasis.opendocument.chart					odc
 application/vnd.oasis.opendocument.database					odb
-- 
2.39.1

